core/sql/exp/ExpError.cpp

```diff
@@ -135,6 +135,20 @@ ComDiagsArea *ExRaiseSqlError(CollHeap* heap, ComDiagsArea** diagsArea,
 			intParam1, intParam2, intParam3, 
 			stringParam1, stringParam2, stringParam3);
 }
+ 
+ComDiagsArea *ExRaiseSqlError(CollHeap* heap, ComDiagsArea** diagsArea,
```

Comment:
In many other places we use ComDiagsArea *&diags, so I'm wondering why we are using ** here.

Reply:
This function is modeled after other ExRaiseSqlError functions already in this file.  I wasn't sure how ComDiagsArea would be passed to the callers of ExRaiseSqlError, meaning if it would exist or not.  If the ComDiagsArea is passed to the caller as a parameter to itself and then it would be become recursive change to ensure that all the function in the chain passes reference pointers.

 The existing ExRaiseSqlError excepts the enum ExeErrorCode needs to be updated to raise sql error correctly.  However, with the embedded arkcmp concepts, even the non-executor error codes needs to be passed to this function.  Casting to enum ExeErrorCode when the code doesn't exist in it causes unexpected value. Hence this new function was introduced to raise sql error. Earlier, the code assumed that diagsArea is always allocated and populated it.

Reply:
Thanks for the explanation.